### PR TITLE
[RHCLOUD-21109] Const AccountNumber refactor

### PR DESCRIPTION
**Pull request by petracihalova**
after discussion in #562 I decided to create PR for each constant from middleware/headers/headers.go 
- to have smaller PRs and 
- to not have chaos in repo commit history

this PR contains changes related with constant ACCOUNT_NUMBER

all PRs related with this constant refactor will be registered in https://github.com/RedHatInsights/sources-api-go/issues/570

**JIRA:** [RHCLOUD-21109](https://issues.redhat.com/browse/RHCLOUD-21109)

### CODE REVIEW

File: util/xrh_header.go [hunk 1]
```diff
@@ -12,8 +12,7 @@ import (
 )
 
 const (
-	xrhAccountNumberKey string = "x-rh-sources-account-number"
-	xrhIdentityKey      string = "x-rh-identity"
+	xrhIdentityKey string = "x-rh-identity"
```
Comment (MikelAlejoBR) [inline util/xrh_header.go]: I would suggest:

```diff
- xrhIdentityKey string = "x-rh-identity"
+ xRhIdentity string = "x-rh-identity"
```

**Review by MikelAlejoBR — APPROVED** [review]

File: util/xrh_header.go (hunk 1 above)
Comment (petracihalova) [reply util/xrh_header.go]: in this PR is solved constant related with account number and that is why here is removed only xrhAccountNumberKey and replaced by constant from middleware/headers/headers.go ... the xrhIdentityKey constant will be replaced here #579

Comment (MikelAlejoBR) [reply util/xrh_header.go]: Oh, I see, thanks!

**Comment by petracihalova** [comment]
rebased & conflict solved

**Review by Ellen-Yi-Dong — APPROVED** [review]

**Comment by petracihalova** [comment]
rebased & conflict solved

**Comment by petracihalova** [comment]
rebased & conflict solved

**Review by lindgrenj6 — APPROVED** [review]

**Pull request merged**